Placement new can have multiple definitions

If the following is added to main.cc:
```diff
    TestHeap_( );

+   std::vector<int> v;
+   v.push_back( 0x56 );
+   v.push_back( 0x44 );
+
    REPORT( "Tests successful!" );
```
Compiling for debug results in a linker error for a multiple definition of `operator new(unsigned long long, void*)`.

--- a/include/egg/eggHeap.hh
+++ b/include/egg/eggHeap.hh
@@ -2,6 +2,8 @@
 
 #include <sdk/heapCommon.hh>
 
+#include <new>
+
 namespace EGG
 {
 
@@ -83,7 +85,6 @@ protected:
 void *operator new( size_t size ) noexcept;
 void *operator new( size_t size, int align ) noexcept;
 void *operator new( size_t size, EGG::Heap *heap, int align ) noexcept;
-void *operator new( size_t size, void *block ) noexcept;
 void *operator new[]( size_t size ) noexcept;
 void *operator new[]( size_t size, int align ) noexcept;
 void *operator new[]( size_t size, EGG::Heap *heap, int align ) noexcept;

--- a/src/egg/eggHeap.cc
+++ b/src/egg/eggHeap.cc
@@ -224,11 +224,6 @@ void *operator new( size_t size, EGG::Heap *heap, int align ) noexcept
     return EGG::Heap::alloc( size, align, heap );
 }
 
-void *operator new( size_t /* size */, void *block ) noexcept
-{
-    return block;
-}
-
 void *operator new[]( size_t size ) noexcept
 {
     return EGG::Heap::alloc( size, 4, nullptr );

--- a/src/main.cc
+++ b/src/main.cc
@@ -1,5 +1,7 @@
 #include <egg/eggExpHeap.hh>
 
+#include <vector>
+
 static void TestBasic_( )
 {
     auto *rootHeap = EGG::ExpHeap::getRootHeap( );
@@ -94,6 +96,19 @@ static void TestHeap_( )
     ASSERT( spacePre == spacePost );
 }
 
+static void TestUse_( )
+{
+    // Check if std::vector reallocates correctly
+    std::vector<int> v;
+    v.push_back( 0x55 );
+    v.push_back( 0xaa );
+    v.push_back( 0x38 );
+    v.push_back( 0x2d );
+
+    ASSERT( GetAddrNum( v.data( ) ) > GetAddrNum( EGG::Heap::getMemorySpace( ) ) );
+    ASSERT( GetAddrNum( v.data( ) ) < GetAddrNum( EGG::Heap::getMemorySpace( ) ) + 0x1000000 );
+}
+
 int main( )
 {
 #ifdef COMMENT
@@ -107,6 +122,7 @@ int main( )
 
     TestBasic_( );
     TestHeap_( );
+    TestUse_( );
 
     REPORT( "Tests successful!" );
 }

--- a/src/sdk/mem_expHeap.cc
+++ b/src/sdk/mem_expHeap.cc
@@ -1,8 +1,7 @@
 #include <sdk/expHeap.hh>
 
-#include <egg/eggHeap.hh> // placement new
-
 #include <limits>
+#include <new> // placement new
 
 namespace RVL
 {

